### PR TITLE
[develop2] New serialization json and printing for ``conan list``

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Feature: New serialization json and printing for ``conan list``.
Docs: https://github.com/conan-io/docs/pull/XXXX

Close https://github.com/conan-io/conan/issues/12804

### CODE REVIEW

File: conans/test/integration/command_v2/list_test.py [hunk 1]
```diff
@@ -2,600 +2,547 @@
...
+        expected = textwrap.dedent("""\
+          conf
+            conf/1.0
+              revisions
+                e4e1703f72ed07c15d73a555ec3a2fa1 (10-11-2023 10:13:13)
+                  packages
+                    78c6fa29e8164ce399087ad6067c8f9e2f1c4ad0
+                      info
+                        conf
+                          tools.build:cxxflags: ['--flag1']
```
Comment (franramirez688) [inline conans/test/integration/command_v2/list_test.py:313]: IMO, I know that this output will have colors, but without `:` or even `[]` to distinguish what are the sections it's a bit difficult for me to read all the information.

File: conan/api/model.py [hunk 2]
```diff
@@ -81,26 +83,31 @@ def prefs(self):
...
+                rev_dict = revisions_dict.setdefault(ref.revision, {})
+                if ref.timestamp:
+                    rev_dict["timestamp"] = timestamp_to_str(ref.timestamp)
+                if prefs:
+                    packages_dict = rev_dict.setdefault("packages", {})
+                    for pref_info in prefs:
+                        pref, info = pref_info
+                        pid_dict = packages_dict.setdefault(pref.package_id, {})
+                        if info is not None:
+                            pid_dict["info"] = info
```
Comment (franramirez688) [inline conan/api/model.py:105]: Why could `info` not be an empty `dict` by default?

I saw this line above:

```python
binary_info = configurations.get(pref) if configurations is not None else None
```
So, I don't know if I'm missing anything else about any caveats.

**Review by franramirez688 — APPROVED** [review]
Overall, I think it looks really good! Great job!

File: conan/api/model.py (hunk 2 above)
Comment (memsharded) [reply conan/api/model.py:105]: I am trying to distinguish the case where the ``info`` is retrieved, even when it is empty, from the cases where the ``info`` is not retrieved at all (like only listing ``prevs`` of a known package_id). ``info=None`` means it is not retrieved at all, while ``info={}`` happens for example for header-only packages which information has been retrieved.

File: conans/test/integration/command_v2/list_test.py (hunk 1 above)
Comment (memsharded) [reply conans/test/integration/command_v2/list_test.py:313]: Yes, I agree. We can format this with whatever syntax we want, this is a proposal for the "structure", I think if we agree on it, we can polish the output format details very easily, lets talk about what we want and I'll do the changes in a following PR.

File: conan/api/model.py (hunk 2 above)
Comment (franramirez688) [reply conan/api/model.py:105]: That's what I wanted to know. Thanks for the clarification! 😉

**Review by AbrilRBS — APPROVED** [review]

**Review by jcar87 — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**